File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  let str = s.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
//   console.log(str);
  return str === str.split("").reverse().join("");
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (Character.isLetter(s.charAt(i))|| Character.isDigit(s.charAt(i))) {
                str.append(s.charAt(i));
            }
        }
        System.out.println(str.toString()); // Output: HelloWorld
        String clean = str.toString().toLowerCase();
        String rev =new StringBuilder (clean).reverse().toString().toLowerCase();
        return clean.equals(rev);
    }
}